docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md

```diff
@@ -15,66 +15,211 @@ author: mikejo5000
+Use the following elements to set up data-driven unit tests. For more information, see [Create a data-driven unit test](../test/how-to-create-a-data-driven-unit-test.md) and [Use a configuration file to define a data source](../test/walkthrough-using-a-configuration-file-to-define-a-data-source.md).
 
-## Attributes used to identify test classes and methods
+- <xref:Microsoft.VisualStudio.TestTools.UnitTesting.DataTestMethodAttribute>
 
-Every test class must have the `TestClass` attribute, and every test method must have the `TestMethod` attribute. For more information, see [Anatomy of a unit test](/previous-versions/ms182517(v=vs.110)).
+- <xref:Microsoft.VisualStudio.TestTools.UnitTesting.DataRowAttribute>
 
-- <xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestClassAttribute>
+- <xref:Microsoft.VisualStudio.TestTools.UnitTesting.DynamicDataAttribute>
 
-- <xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestMethodAttribute>
+- <xref:Microsoft.VisualStudio.TestTools.UnitTesting.DataSourceAttribute>
 
-## Assert classes and related exceptions
+### Attributes used to provide initialization and cleanups
+
+A method decorated with one of the following attributes is called at the moment you specify. For more information, see [Anatomy of a unit test](/previous-versions/ms182517(v=vs.110)).
+
+#### Assembly
+
+[AssemblyInitialize](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.AssemblyInitializeAttribute>) is called right after your assembly is loaded and [AssemblyCleanup](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.AssemblyCleanupAttribute>) is called right before your assembly is unloaded.
+
+The methods marked with these attributes should be defined as `static void` or `static Task`, in a `TestClass`, and appear only once. The initialize part requires one argument of type [TestContext](xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestContext) and the cleanup no argument.
+
+```csharp
+[TestClass]
+public class MyTestClass
+{
+    [AssemblyInitialize]
+    public static void AssemblyInitialize(TestContext testContext)
+    {
+    }
+
+    [AssemblyCleanup]
+    public static void AssemblyCleanup()
+    {
+    }
+}
+```
+
+```csharp
+[TestClass]
+public class MyOtherTestClass
+{
+    [AssemblyInitialize]
+    public static async Task AssemblyInitialize(TestContext testContext)
+    {
+    }
+
+    [AssemblyCleanup]
+    public static async Task AssemblyCleanup()
+    {
+    }
+}
+```
+
+#### Class
+
+[ClassInitialize](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.ClassInitializeAttribute>) is called right before your class is loaded (but after static constructor) and [ClassCleanup](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.ClassCleanupAttribute>) is called right after your class is unloaded.
+
+It is possible to control the inheritance behavior: only for current class using `InheritanceBehavior.None` or for all derived classes using `InheritanceBehavior.BeforeEachDerivedClass`.
```

Comment:
```suggestion
It is possible to control the inheritance behavior only for the current class using `InheritanceBehavior.None` or for all derived classes using `InheritanceBehavior.BeforeEachDerivedClass`.
```

Reply:
I am not native English speaker but isn't it changing the meaning of the sentence? I want to say use `None` to have no inheritance and `BeforeEachDerivedClass` to have inheritance.